<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        
        
        
        <link rel="shortcut icon" href="../../img/favicon.ico">
        <title>Annotation - Meta-Omics-Toolkit</title>
        <link href="../../css/bootstrap.min.css" rel="stylesheet">
        <link href="../../css/font-awesome.min.css" rel="stylesheet">
        <link href="../../css/base.css" rel="stylesheet">
        <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/styles/github.min.css">
        <link href="../../css/version-select.css" rel="stylesheet">
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/highlight.min.js"></script>
        <script>hljs.highlightAll();</script> 
    </head>

    <body>
        <div class="navbar fixed-top navbar-expand-lg navbar-dark bg-primary">
            <div class="container">
                <a class="navbar-brand" href="../..">Meta-Omics-Toolkit</a>
                <!-- Expander button -->
                <button type="button" class="navbar-toggler" data-toggle="collapse" data-target="#navbar-collapse">
                    <span class="navbar-toggler-icon"></span>
                </button>

                <!-- Expanded navigation -->
                <div id="navbar-collapse" class="navbar-collapse collapse">
                        <!-- Main navigation -->
                        <ul class="nav navbar-nav">
                            <li class="navitem">
                                <a href="../.." class="nav-link">Home</a>
                            </li>
                            <li class="navitem">
                                <a href="../../pipeline_configuration/" class="nav-link">Global Pipeline Configuration Settings</a>
                            </li>
                            <li class="dropdown active">
                                <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown">Modules <b class="caret"></b></a>
                                <ul class="dropdown-menu">
                                    
<li>
    <a href="../qualityControl/" class="dropdown-item">Quality Control</a>
</li>
                                    
<li>
    <a href="./" class="dropdown-item active">Annotation</a>
</li>
                                    
<li>
    <a href="../assembly/" class="dropdown-item">Assembly</a>
</li>
                                    
<li>
    <a href="../cooccurrence/" class="dropdown-item">Cooccurrence</a>
</li>
                                    
<li>
    <a href="../dereplication/" class="dropdown-item">Dereplication</a>
</li>
                                    
<li>
    <a href="../fragment_recruitment/" class="dropdown-item">Run Fragment Recruitment</a>
</li>
                                    
<li>
    <a href="../magAttributes/" class="dropdown-item">MagAttributes</a>
</li>
                                    
<li>
    <a href="../plasmids/" class="dropdown-item">Plasmids</a>
</li>
                                    
<li>
    <a href="../metabolomics/" class="dropdown-item">Metabolomics</a>
</li>
                                    
<li>
    <a href="../readMapping/" class="dropdown-item">Read Mapping</a>
</li>
                                </ul>
                            </li>
                            <li class="dropdown">
                                <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown">Developer <b class="caret"></b></a>
                                <ul class="dropdown-menu">
                                    
<li>
    <a href="../../developer_guidelines/" class="dropdown-item">Guidelines</a>
</li>
                                    
<li>
    <a href="../../pipeline_specification/" class="dropdown-item">Pipeline Specification</a>
</li>
                                    
<li>
    <a href="../../module_specification/" class="dropdown-item">Module Specification</a>
</li>
                                </ul>
                            </li>
                        </ul>

                    <ul class="nav navbar-nav ml-auto">
                        <li class="nav-item">
                            <a href="#" class="nav-link" data-toggle="modal" data-target="#mkdocs_search_modal">
                                <i class="fa fa-search"></i> Search
                            </a>
                        </li>
                            <li class="nav-item">
                                <a rel="prev" href="../qualityControl/" class="nav-link">
                                    <i class="fa fa-arrow-left"></i> Previous
                                </a>
                            </li>
                            <li class="nav-item">
                                <a rel="next" href="../assembly/" class="nav-link">
                                    Next <i class="fa fa-arrow-right"></i>
                                </a>
                            </li>
                    </ul>
                </div>
            </div>
        </div>

        <div class="container">
            <div class="row">
                    <div class="col-md-3"><div class="navbar-light navbar-expand-md bs-sidebar hidden-print affix" role="complementary">
    <div class="navbar-header">
        <button type="button" class="navbar-toggler collapsed" data-toggle="collapse" data-target="#toc-collapse" title="Table of Contents">
            <span class="fa fa-angle-down"></span>
        </button>
    </div>

    
    <div id="toc-collapse" class="navbar-collapse collapse card bg-secondary">
        <ul class="nav flex-column">
            
            <li class="nav-item" data-level="1"><a href="#annotation" class="nav-link">Annotation</a>
              <ul class="nav flex-column">
            <li class="nav-item" data-level="2"><a href="#input" class="nav-link">Input</a>
              <ul class="nav flex-column">
              </ul>
            </li>
            <li class="nav-item" data-level="2"><a href="#output" class="nav-link">Output</a>
              <ul class="nav flex-column">
              </ul>
            </li>
              </ul>
            </li>
        </ul>
    </div>
</div></div>
                    <div class="col-md-9" role="main">

<h1 id="annotation">Annotation</h1>
<p>The annotation module is able to predict genes and annotate those based on Prokka and a set of user provided databases.
A user can add additional formatted databases as part of the configuration by adding a key (Example: <code>kegg</code> ) with 
a possible download strategy. See <a href="../../pipeline_configuration/#database-input-configuration">database section</a> for possible download strategies.
In addition, the resistance gene identifier is executed by default.</p>
<h2 id="input">Input</h2>
<div class="tabbed-set tabbed-alternate" data-tabs="1:3"><input checked="checked" id="__tabbed_1_1" name="__tabbed_1" type="radio" /><input id="__tabbed_1_2" name="__tabbed_1" type="radio" /><input id="__tabbed_1_3" name="__tabbed_1" type="radio" /><div class="tabbed-labels"><label for="__tabbed_1_1">Command</label><label for="__tabbed_1_2">Configuration File</label><label for="__tabbed_1_3">TSV Table</label></div>
<div class="tabbed-content">
<div class="tabbed-block">
<div class="highlight"><pre><span></span><code>-entry wAnnotateLocal -params-file example_params/annotation.yml
</code></pre></div>
</div>
<div class="tabbed-block">
<div class="highlight"><pre><span></span><code>
</code></pre></div>
</div>
<div class="tabbed-block">
<div class="highlight"><pre><span></span><code>
</code></pre></div>
</div>
</div>
</div>
<h3 id="databases">Databases</h3>
<h3 id="mmseqs2">MMseqs2</h3>
<p>MMseqs2 needs a combination of different data, index and dbtype files as "one" database, be it in- or output.
See <a href="https://github.com/soedinglab/mmseqs2/wiki#mmseqs2-database-format">MMseqs2 database</a> for more information.
As multiple and in most cases, big files are used, tar and <a href="https://github.com/facebook/zstd">zstd</a> are utilized to compress and transport files.
Input databases have to be compressed by these and need to end with <code>.tar.zst</code>. Naming inside an archive is irrelevant, as databases are picked automatically.
Multiple databases per one archive are not supported, one archive, one database. If the database also includes a taxonomy 
as described <a href="https://github.com/soedinglab/mmseqs2/wiki#creating-a-seqtaxdb">here</a>, it can also be used for taxonomic classifications with MMseqs2 - Taxonomy.
See <a href="../../pipeline_configuration/#database-input-configuration">database section</a> for possible download strategies.
If you need credentials to access your files via S3 then please use the following command:</p>
<div class="highlight"><pre><span></span><code>nextflow secrets set S3_db_ACCESS XXXXXXX
nextflow secrets set S3_db_SECRET XXXXXXX
</code></pre></div>
<p>where <code>db</code> is the name of the database that you use in your config file.
Example:</p>
<div class="highlight"><pre><span></span><code>....
      vfdb:
        params: &#39; -s 1 --max-seqs 100 --max-accept 50 --alignment-mode 1 --exact-kmer-matching 1 --db-load-mode 3&#39;
        database:
          download:
            source: s3://databases/vfdb_full_2022_07_29.tar.zst
            md5sum: 7e32aaed112d6e056fb8764b637bf49e
            s5cmd:
              params: &quot; --retry-count 30 --endpoint-url https://openstack.cebitec.uni-bielefeld.de:8080 &quot; 
....
</code></pre></div>
<p>Based on these settings, you would set the following secret:</p>
<div class="highlight"><pre><span></span><code>nextflow secrets set S3_vfdb_ACCESS XXXXXXX
nextflow secrets set S3_vfdb_SECRET XXXXXXX
</code></pre></div>
<h3 id="keggfromblast">KEGGFromBlast</h3>
<p>KeGGFromBlast is only executed if genes are searched against a KEGG database. There must be a <code>kegg</code> identifier (see example configuration file) in the annotation section.
KeGGFromBlast needs a kegg database as input which must be a tar.gz file.
See <a href="../../pipeline_configuration/#database-input-configuration">database section</a> for possible download strategies.
If you need credentials to access your files via S3 then please use the following command:</p>
<div class="highlight"><pre><span></span><code>nextflow secrets set S3_kegg_ACCESS XXXXXXX
nextflow secrets set S3_kegg_SECRET XXXXXXX
</code></pre></div>
<h3 id="mmseqs-taxonomy">MMSeqs Taxonomy</h3>
<p>If you need credentials to access your files via S3 then please use the following command:</p>
<div class="highlight"><pre><span></span><code>nextflow secrets set S3_TAX_db_ACCESS XXXXXXX
nextflow secrets set S3_TAX_db_SECRET XXXXXXX
</code></pre></div>
<p>where <code>db</code> is the name of the database that you use in your config file.
Example:</p>
<div class="highlight"><pre><span></span><code>....
    mmseqs2_taxonomy:
      gtdb:
        params: &#39; --orf-filter-s 1 -e 1e-15&#39;
        ramMode: false
        database:
          download:
            source: s3://databases/gtdb_r214_1_mmseqs.tar.gz
            md5sum: 3c8f12c5c2dc55841a14dd30a0a4c718
            s5cmd:
              params: &quot; --retry-count 30 --endpoint-url https://openstack.cebitec.uni-bielefeld.de:8080 &quot; 
....
</code></pre></div>
<p>Based on these settings, you would set the following secrets:</p>
<div class="highlight"><pre><span></span><code>nextflow secrets set S3_TAX_gtdb_ACCESS XXXXXXX
nextflow secrets set S3_TAX_gtdb_SECRET XXXXXXX
</code></pre></div>
<h3 id="rgi">RGI</h3>
<p>RGI needs a CARD database which can be fetched via this link:  https://card.mcmaster.ca/latest/data.
The compressed database must be a tar.bz2 file. 
See <a href="../../pipeline_configuration/#database-input-configuration">database section</a> for possible download strategies.
If you need credentials to access your files via S3 then please use the following command:</p>
<div class="highlight"><pre><span></span><code>nextflow secrets set S3_rgi_ACCESS XXXXXXX
nextflow secrets set S3_rgi_SECRET XXXXXXX
</code></pre></div>
<h2 id="output">Output</h2>
<h3 id="mmseqs2_1">MMseqs2</h3>
<p>Calculated significant matches of a nucleotide/protein query which was compared against a user provided set of databases.</p>
<h3 id="mmseqs2-taxonomy">MMseqs2 - Taxonomy</h3>
<p>By identifying homologous through searches against a provided MMseqs2 taxonomy-database, MMseqs2 can compute the lowest common ancestor. 
This lowest common ancestor is a robust taxonomic label for unknown sequences.
These labels are presented in form of an <code>*.taxonomy.tsv</code> file, a <code>*.krakenStyleTaxonomy.out</code> formatted in accordance to the <a href="https://ccb.jhu.edu/software/kraken/">KRAKEN tool</a> outputs and
an interactive <a href="https://github.com/marbl/Krona/wiki">KRONA</a> plot in form of a html website <code>*.krona.html</code>.</p>
<h3 id="prokka">Prokka</h3>
<p>Prokka computes <code>*.err</code>, <code>*.faa</code>, <code>*.ffn</code>, <code>*.fna</code>, <code>*.fsa</code>, <code>*.gbk</code>, <code>*.gff</code>, <code>*.sqn</code>, <code>*.tbl</code>, <code>*.tbl</code> for every bin.
<code>*.gbk</code> and <code>*.sqn</code> are skipped per default, since tbl2asn runs for quite a while! If you need those files generated by prokka, include:
<code>--tbl2asn</code> in the prokka parameters to enable it.
Details of all files can be read on the Prokka page.
In addition, it also computes a summary tsv file which adheres to the magAttributes specification.</p>
<h3 id="keggfromblast_1">KEGGFromBlast</h3>
<p>Result <code>*.tsv</code> file filled with KEGG information (like modules, KO's, ...) which could be linked to the input hits.</p>
<h3 id="resistance-gene-identifier-rgi">Resistance Gene Identifier (rgi)</h3>
<p>The <code>*rgi.tsv</code> files contain the found CARD genes.</p></div>
            </div>
        </div>

        <footer class="col-md-12">
            <hr>
            <p>Documentation built with <a href="https://www.mkdocs.org/">MkDocs</a>.</p>
        </footer>
        <script src="../../js/jquery-3.6.0.min.js"></script>
        <script src="../../js/bootstrap.min.js"></script>
        <script>
            var base_url = "../..",
                shortcuts = {"help": 191, "next": 78, "previous": 80, "search": 83};
        </script>
        <script src="../../js/base.js"></script>
        <script src="../../search/main.js"></script>
        <script src="../../js/version-select.js"></script>

        <div class="modal" id="mkdocs_search_modal" tabindex="-1" role="dialog" aria-labelledby="searchModalLabel" aria-hidden="true">
    <div class="modal-dialog modal-lg">
        <div class="modal-content">
            <div class="modal-header">
                <h4 class="modal-title" id="searchModalLabel">Search</h4>
                <button type="button" class="close" data-dismiss="modal"><span aria-hidden="true">&times;</span><span class="sr-only">Close</span></button>
            </div>
            <div class="modal-body">
                <p>From here you can search these documents. Enter your search terms below.</p>
                <form>
                    <div class="form-group">
                        <input type="search" class="form-control" placeholder="Search..." id="mkdocs-search-query" title="Type search term here">
                    </div>
                </form>
                <div id="mkdocs-search-results" data-no-results-text="No results found"></div>
            </div>
            <div class="modal-footer">
            </div>
        </div>
    </div>
</div><div class="modal" id="mkdocs_keyboard_modal" tabindex="-1" role="dialog" aria-labelledby="keyboardModalLabel" aria-hidden="true">
    <div class="modal-dialog">
        <div class="modal-content">
            <div class="modal-header">
                <h4 class="modal-title" id="keyboardModalLabel">Keyboard Shortcuts</h4>
                <button type="button" class="close" data-dismiss="modal"><span aria-hidden="true">&times;</span><span class="sr-only">Close</span></button>
            </div>
            <div class="modal-body">
              <table class="table">
                <thead>
                  <tr>
                    <th style="width: 20%;">Keys</th>
                    <th>Action</th>
                  </tr>
                </thead>
                <tbody>
                  <tr>
                    <td class="help shortcut"><kbd>?</kbd></td>
                    <td>Open this help</td>
                  </tr>
                  <tr>
                    <td class="next shortcut"><kbd>n</kbd></td>
                    <td>Next page</td>
                  </tr>
                  <tr>
                    <td class="prev shortcut"><kbd>p</kbd></td>
                    <td>Previous page</td>
                  </tr>
                  <tr>
                    <td class="search shortcut"><kbd>s</kbd></td>
                    <td>Search</td>
                  </tr>
                </tbody>
              </table>
            </div>
            <div class="modal-footer">
            </div>
        </div>
    </div>
</div>

    </body>
</html>
